### PR TITLE
Convert to es6 & Documentation

### PR DESCRIPTION
**Pull request by sagspot**

### CODE REVIEW

File: controllers/parcels.js [hunk 1]
```diff
@@ -1,25 +1,29 @@
...
-    (parcel) => parcel.toUser == req.user.email
-  );
-  res.json({ sentParcels, receivedParcels });
+  try {
+    const receivedParcels = parcels.filter(
+      (parcel) => parcel.toUser == req.user.email
+    );
+    res.json({ sentParcels, receivedParcels });
+  } catch (err) {
+    res.status(500).send('Something went wrong');
```
Comment (Nicanor008) [inline controllers/parcels.js]: Return something in try catch
```suggestion
  try {
    const receivedParcels = parcels.filter(
      (parcel) => parcel.toUser == req.user.email
    );
    return res.json({ sentParcels, receivedParcels });
  } catch (err) {
    return res.status(500).send('Something went wrong');
```

File: controllers/parcels.js [hunk 2]
```diff
@@ -1,25 +1,29 @@
...
+import moment from 'moment';
 
-const {
+import {
   sendParcelValidation,
   receiveParcelValidation,
-} = require('../middlewares/validation');
-const parcels = require('../model/parcels');
+} from '../middlewares/validation.js';
+import parcels from '../model/parcels.js';
```
Comment (Nicanor008) [inline controllers/parcels.js:8]: Not sure why you adding the `.js`, it should automatically detect file type and import

File: controllers/parcels.js [hunk 3]
```diff
@@ -1,25 +1,29 @@
...
+  } catch (err) {
+    res.status(500).send('Something went wrong');
+  }
 };
 
-exports.parcel_post_send = (req, res) => {
+export const parcel_post_send = (req, res) => {
   // Validate parcel
   const { error } = sendParcelValidation(req.body);
   if (error) return res.status(400).send(error.details[0].message);
```
Comment (Nicanor008) [inline controllers/parcels.js]: This is a bad pattern, prone to errors, and not scalable especially on data indexing change,
either just error only or details without the `[0]`

File: controllers/parcels.js [hunk 4]
```diff
@@ -46,7 +50,9 @@ exports.parcel_post_send = (req, res) => {
   }
 };
 
-exports.parcel_patch_receive = (req, res) => {
+export const parcel_patch_receive = (req, res) => {
+  if (!req.params.id) return res.status(400).send('ID is missing in the URI');
+
   // Validate parcel
   const { error } = receiveParcelValidation(req.body);
   if (error) return res.status(400).send(error.details[0].message);
```
Comment (Nicanor008) [inline controllers/parcels.js]: same here

File: controllers/users.js [hunk 5]
```diff
@@ -22,37 +22,42 @@ exports.users_post_register = async (req, res) => {
...
     password: req.body.password,
   };
 
   try {
     await users.push(newUser);
-    res.send(newUser);
+    res.json({
+      id: newUser.id,
+      name: newUser.name,
+      email: newUser.email,
```
Comment (Nicanor008) [inline controllers/users.js]: How about you destructure this
```suggestion
     const { id, name, email } = newUser;
      id,
      name,
      email,
```

**Review by Nicanor008 — CHANGES_REQUESTED** [review]
Nice stuff, just check on this improvements

File: controllers/parcels.js (hunk 2 above)
Comment (sagspot) [reply controllers/parcels.js:8]: It doesn't, maybe that's the downside of using `{type: modules}`  in the package.json

**Pull request merged**